inc/karray.h

```diff
@@ -0,0 +1,865 @@
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+
+/*++
+
+Abstract:
+
+    Implements a C++ container analogous to std::vector
+
+Environment:
+
+    Kernel mode or usermode unittest
+
+Notes:
+
+    Because kernel C++ doesn't support exceptions, we can't use the STL directly
+    in kernelmode.  Therefore, this class provides a limited and slightly
+    modified subset of the STL's std::vector.
+
+    If you're not familiar with std::vector, you should go read up on it
+    first: https://docs.microsoft.com/en-us/cpp/standard-library/vector-class
+
+    This file was originally copied from the following location and then
+    modified to reduce dependencies:
+
+    https://github.com/microsoft/Network-Adapter-Class-Extension/blob/windows_10.0.19541/ndis/rtl/inc/karray.h
+
+--*/
+
+#pragma once
+
+#include "cxplat.h"
+#include <new.h>
+#ifdef KERNEL_MODE
+#include <ntintsafe.h>
+#else
+#include <intsafe.h>
+#define RtlSIZETMult SizeTMult
+#endif
+#include <wil/wistd_type_traits.h>
```

Comment:
Why does this depend on wil? The MsQuic version doesn't (it doesn't depend on wil at all). I believe MsQuic simply copied the few dependencies locally into this file, which would still be my preference instead of a platform specific dependency.

Reply:
removed wil dependency